### DISCUSSION TITLE
How to make supabase auth work in add to home screen iOS

### DISCUSSION THREAD

**Discussion by gokulkrishh** [supabase/supabase#12227] [Questions]
BG: I have created an expense tracker called [expense.fyi](https://expense.fyi), its OSS and code available is [github](https://github.com/gokulkrishh/expense.fyi).

It uses supabase auth magic link to sign up/sign in users.

Problem i am facing now is, in iOS after adding to home screen, app is not in logged in state even though its logged in safari, i believe its due to how safari works after adding to home screen.

Note: It works perfectly fine in android after adding to home screen, user stays logged in.

Question is: How to make it work after adding to home screen in standalone mode? Has anyone solved this problem before?

Thanks in advance.

**Comment by GaryAustin1** [comment]
I have a PWA and solved it by using OTP token instead of a link.   This way the user stays in the homescreen app when signing in.
But I expect the user to already be in the homescreen app when they signin, not transition from safari to the new home screen app after.

**Comment by gokulkrishh** [comment]
@GaryAustin1 Thanks, yes that should work. But for my use case, i would like to keep the magic link and make it work as it is now. Will wait for some others to post they replies.

**Reply by jareddr** [reply]
Did you ever solve this problem?  I have the same question.

**Reply by gokulkrishh** [reply]
@jareddr Only way to solve this is to use OTP via SMS or email. No other way.

**Reply by GaryAustin1** [reply]
There is this method in the docs.  https://supabase.com/docs/guides/auth/auth-email-templates#email-prefetching

**Comment by justman00** [comment]
Anybody knows how to handle this in case of `Login with Google | Apple | yada | yada`? A browser popup is appearing, however it is not connected to any session that the user has, so in order to login with google, the user needs to log into their google account as if they were on a new device, which is very annoying

**Comment by bernatfortet** [comment]
Did anyone figure this one out?
